src/pages/ProductListing/components/ProductCard/ProductCard.jsx

```diff
@@ -0,0 +1,104 @@
+// All module imports
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+// All local imports
+import "./ProductCard.css";
+import { useProducts } from "../../../../context/products-context";
+import { displayRatingStars } from "../../../../utils/displayRatingStars";
+
+export function ProductCard({ product }) {
+    const {
+        imageSrc,
+        imageAlt,
+        bestSeller,
+        title,
+        description,
+        price,
+        percentDiscount,
+    } = product;
+
+    const navigate = useNavigate();
+    const { wishlist, dispatch } = useProducts();
+    const [wishlistLoading, setWishlistLoading] = useState(false);
+
+    const isAddedToWishlist = () =>
+        wishlist.some((obj) => obj._id === product._id);
+
+    const wishlistIcon = () => {
+        return isAddedToWishlist() ? (
+            <i className="material-icons card-svg-icons wishlist-icon">
+                favorite
+            </i>
+        ) : (
+            <i className="material-icons card-svg-icons">favorite_border</i>
+        );
+    };
+
+    return (
+        <li
+            className="card-item"
+            onClick={() => navigate(`/product-detail?id=${product._id}`)}
+        >
+            <div className="card card-vertical card-with-badge">
+                <div className="card-body">
+                    <div className="card-image-header-container">
+                        <img
+                            src={imageSrc}
+                            alt={imageAlt}
+                            className="card-image"
+                        />
+                        {bestSeller ? (
+                            <span className="card-badge">Bestseller</span>
+                        ) : (
+                            ""
+                        )}
+                        <div className="card-secondary-buttons-container">
+                            <button
+                                className="card-button"
+                                onClick={(e) => {
+                                    e.stopPropagation();
```

Comment:
define this as a function before the jsx , don't put complicated or multi liner  functions in the jsx itself , damages the code readibility

Reply:
Ok done